File: source/nestjs-petclinic-rest/src/services/dto/pet-update.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { Type } from 'class-transformer';

export class PetUpdateDTO {
  @ApiProperty()
  name: string;

  @ApiProperty()
  @Type(() => Date)
  birthDate: Date;

  @ApiProperty()
  petTypeId: number;
}

File: source/dotnet-petclinic-rest/PetClinic.Application/Dtos/PetUpdateDTO.cs
using System;
using System.Collections.Generic;
using Intent.RoslynWeaver.Attributes;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Dtos.DtoModel", Version = "1.0")]

namespace PetClinic.Application.Dtos
{

    public class PetUpdateDTO
    {
        public PetUpdateDTO()
        {
        }

        public static PetUpdateDTO Create(string name, DateTime birthDate, int petTypeId)
        {
            return new PetUpdateDTO
            {
                Name = name,
                BirthDate = birthDate,
                PetTypeId = petTypeId
            };
        }

        public string Name { get; set; }

        public DateTime BirthDate { get; set; }

        public int PetTypeId { get; set; }

    }
}